### PR TITLE
New Feature/actions file menu

### PR DESCRIPTION
**Pull request by Abijeet**
Towards: https://github.com/nextcloud/server/issues/7647

As described, have created a new drop-down menu shown when multiple files are selected. I also modified the trashbin plugin (Deleted files), I've put the restore and delete items under the actions menu.

Here are a few screenshots,

## Laptop

screenshot from 2018-06-02 21-02-40

### Deleted files

screenshot from 2018-06-02 21-29-33

### Landscape
localhost_nextcloud_index php_apps_files__dir _ iphone 6_7_8 plus

### Potrait
localhost_nextcloud_index php_apps_files__dir _ iphone 6_7_8 plus 1

### CODE REVIEW

File: core/css/mobile.scss [hunk 1]
```diff
@@ -67,7 +67,7 @@
 #app-navigation-toggle {
 	position: fixed;
 	display: inline-block !important;
-	top: 45px;
+	top: 50px;
```
Comment (Abijeet) [inline core/css/mobile.scss]: This change was made to fix an issue I noticed with the navigation toggle on mobile resolutions,

## Without fix
localhost_nextcloud_index php_apps_files__dir _ view trashbin pixel 2 xl

## With fix

localhost_nextcloud_index php_apps_files__dir _ view trashbin pixel 2 xl 1

**Review by jancborchardt — APPROVED** [review]

**Comment by jancborchardt** [comment]
This looks great! :) Please also review @nextcloud/designers @nextcloud/javascript

**Comment by skjnldsv** [comment]
@jancborchardt not sure it's ready to review since @Abijeet left the WIP in the title. :)

File: core/css/mobile.scss (hunk 1 above)
Comment (jancborchardt) [reply core/css/mobile.scss]: Thanks, good catch! That’s because we increased the height of the header recently. :)

**Review by jancborchardt — COMMENTED** [review]
-

**Comment by jancborchardt** [comment]
@skjnldsv hmm, can’t seem to withdraw the review? :\

**Comment by skjnldsv** [comment]
@jancborchardt you need to request changes indeed. You cannot 'remove' the review :)

**Comment by Abijeet** [comment]
I'll work on fixing the test cases.

**Comment by MorrisJobke** [comment]
Please rebase on the latest master to get the fixes for the mysql failures ;)

**Comment by Abijeet** [comment]
> Please rebase on the latest master to get the fixes for the mysql failures ;)

@MorrisJobke  - Thanks. I've done that and that seems to have fixed the MySQL failures. Not sure about the other failing issue, I think it's not related to my changes

**Comment by rullzer** [comment]
@Abijeet since you changed the title and the tests pass I assume this is good to review

**Comment by rullzer** [comment]
@juliushaertl @skjnldsv @danxuliu work your review magic!

**Comment by juliushaertl** [comment]
@Abijeet Really nice improvement :+1: 

Can you have a look at the positioning of the text when selecting an element, it seems to be some pixels above the text/checkbox when no element is selected.

peek 2018-06-11 09-36

**Comment by Abijeet** [comment]
Hey @juliushaertl,
> Can you have a look at the positioning of the text when selecting an element, it seems to be some pixels above the text/checkbox when no element is selected.

I'm sorry I noticed this and it subsequently skipped my mind. I'll get round to fixing it soon.

**Comment by Abijeet** [comment]
I've fixed the issues with vertical moving. I'm also noticing that the name column expands slightly when a row is selected. Its because the Size column on the top which contains the size of the file (that appears when a file is selected) is wider when compared to the column heading "Size". 

We also have a hidden (not `display: none`) sort icon next to the file size which if you notice is causing the total size not to be aligned with the individual file size options. I assume we used `visibility: hidden` instead of `display: none` in order to not disturb the size of the columns on selection.

sideway-movings

**Comment by Abijeet** [comment]
Hey @juliushaertl 

This is ready for another review. I've made some other UI tweaks since you last reviewed it. 

I also spent some time on the issue below but could not fix it,

sideway-movings

I'm fairly confident that this was not caused due to my changes so I think we can go ahead and merge this if you guys think its ready.

Let me know if you want me to create another issue to handle the bug in the GIF.

**Review by skjnldsv — APPROVED** [review]

**Comment by MorrisJobke** [comment]
I fixed the alignment of the more icon:

Before:

bildschirmfoto 2018-06-15 um 17 08 57

After:

bildschirmfoto 2018-06-15 um 17 08 53

There is one little glitch remaining:

* [ ] when there is no file selected the actions menu is visible behind the file list
bildschirmfoto 2018-06-15 um 17 13 16

* [ ] when start doing a move and then abort the dialog (instead of click move and copy) then the loading circle is still present in the menu

Beside that it looks really good 👍

**Comment by Abijeet** [comment]
Hey @MorrisJobke,

Thanks for checking this out.

>  when there is no file selected the actions menu is visible behind the file list

I should have noticed this. It was caused due to my latest changes. Fixed it.

>  when start doing a move and then abort the dialog (instead of click move and copy) then the loading circle is still present in the menu

This is also fixed. A similar issue was present on the Deleted files restore icon as well, which I have fixed too.

I've added some more assertions to avoid some of the issues from happening.

**Comment by Abijeet** [comment]
Re-basing with the master branch does not seem to have fixed the build failures.

Build failing - **TESTS=vue-builds**

Error,
```
./build/vue-builds.sh ./settings/js/main.js
The build file ./settings/js/main.js does not exists 
```

**Comment by skjnldsv** [comment]
> This branch is 10 commits ahead, 54 commits behind master.
I guess the rebase failed :)

**Pull request merged**